### PR TITLE
Default deploy chainmail

### PR DESCRIPTION
**Pull request by James-Mart**
This PR ensures that the ChainMail app is deployed to the chain by default.

To accomplish this, the [new functionality](https://github.com/gofractally/psibase/pull/840) in `cargo-psibase` that allows for UI and postinstall scripts to be specified is leveraged in chainmail.

Also, this PR introduces a new CMake function `cargo_psibase_package` that allows psibase devs to tell CMake about packages that are built using `cargo-psibase package` rather than using the CMake `psibase_package` function. Therefore cargo-psibase-managed packages can be integrated into the regular build system just like the other packages.

### CODE REVIEW

File: doc/psidk/src/development/services/rust-service/package.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ A [psibase package](../../../specifications/data-formats/package.md) may be comp
 The available fields are:
 
 - `package-name`: Must be present on the top-level crate for the package.
+- `description`: A description of the package.
```
Comment (swatanabe) [inline doc/psidk/src/development/services/rust-service/package.md:10]: `description` and `version` are taken from `[package]`, not `[package.metadata.psibase]`

Comment (James-Mart) [reply doc/psidk/src/development/services/rust-service/package.md:10]: Oh. I'll remove this and also move the description field in the `test_package` to `[package]`.

Comment (James-Mart) [reply doc/psidk/src/development/services/rust-service/package.md]: Done

File: libraries/psibase/sdk/pack_service.cmake [hunk 2]
```diff
@@ -306,3 +306,47 @@ function(psibase_package)
...
+        INSTALL_COMMAND ""
+        BUILD_ALWAYS 1
+        DEPENDS ${ARG_DEPENDS} cargo-psibase
+    )
+
+    add_custom_command(
+        OUTPUT ${PACKAGE_OUTPUT}
+        COMMAND ${CMAKE_COMMAND} -E true
+        DEPENDS ${TARGET_NAME}_ext
+    )
```
Comment (swatanabe) [inline libraries/psibase/sdk/pack_service.cmake:343]: This is weird. The reason that it was failing without this is that the other `add_custom_command` only has a file level dependency on `${PACKAGE_OUTPUT}` but not a target level dependency on `${TARGET_NAME}_ext`.

Comment (James-Mart) [reply libraries/psibase/sdk/pack_service.cmake]: Fixed

**Review by swatanabe — APPROVED** [review]

**Pull request merged**